src/pages/experiments/[experimentId]/data-exploration/components/heatmap/HeatmapPlot.jsx

```diff
@@ -20,59 +21,43 @@ const HeatmapPlot = (props) => {
   const componentType = 'Heatmap';
 
   const dispatch = useDispatch();
-  const { data: selectedGenes, fetching } = useSelector((state) => state.genes.expression.views[componentType]);
 
-  // const selectedGenes = useSelector((state) => state.genes.expression.views[componentType]?.data);
-  // const selectedGenesLoading = useSelector((state) => state.genes.expression.views[componentType]?.fetching);
+  const loadingGenes = useSelector((state) => state.genes.expression.loading);
+  const selectedGenes = useSelector((state) => state.genes.expression.views[componentType]?.data);
+  const [vegaData, setVegaData] = useState(null);
+
   const expressionData = useSelector((state) => state.genes.expression);
   const hoverCoordinates = useRef({});
-
   const cellSetData = useSelector((state) => state.cellSets);
-
   const { expressionError: error } = expressionData;
   const viewError = useSelector((state) => state.genes.expression.views[componentType]?.error);
 
-  if (!selectedGenes || selectedGenes.length === 0) {
-    return (
-      <center>
-        <Empty
-          description={(
-            <span>
-              Please add gene(s) to the heatmap from the Gene list tool
-            </span>
-          )}
-        />
-        <HeatmapCrossHairs />
-      </center>
-    );
-  }
+  useEffect(() => {
+    console.log(loadingGenes, 'loadingGenes');
+    console.log(selectedGenes, 'selectedGenes');
 
-  if (fetching) {
-    return (
-      <center style={{ marginTop: height / 2 }}>
-        <Spin size='large' />
-        <HeatmapCrossHairs />
-      </center>
-    );
-  }
+    if (!selectedGenes || selectedGenes.length === 0) {
+      console.log('skipping because the selected genes dont exist or are none');
+      return;
+    }
```

Comment:
don't do anything with data if selectedGenes is none or empty -- in this case `createVegaData` would break, so we'd get nothing back. let's leave it at the previous state.